File: app/src/androidTest/java/it/droidcon/testingdaggerrxjava/test6/MyDaggerMockRule.java
package it.droidcon.testingdaggerrxjava.test6;

import it.cosenonjaviste.daggermock.DaggerMockRule;
import it.droidcon.testingdaggerrxjava.dagger.ApplicationComponent;
import it.droidcon.testingdaggerrxjava.dagger.UserInteractorModule;

import static it.droidcon.testingdaggerrxjava.TestUtils.getAppFromInstrumentation;

public class MyDaggerMockRule extends DaggerMockRule<ApplicationComponent> {
  public MyDaggerMockRule() {
    super(
        ApplicationComponent.class,
        new UserInteractorModule()
    );
    set(component ->
        getAppFromInstrumentation().setComponent(component));
  }
}

File: app/src/androidTest/java/it/droidcon/testingdaggerrxjava/test6/MyDaggerMockRule.kt
package it.droidcon.testingdaggerrxjava.test6

import it.cosenonjaviste.daggermock.DaggerMockRule
import it.droidcon.testingdaggerrxjava.appFromInstrumentation
import it.droidcon.testingdaggerrxjava.dagger.ApplicationComponent
import it.droidcon.testingdaggerrxjava.dagger.UserInteractorModule

class MyDaggerMockRule : DaggerMockRule<ApplicationComponent>(ApplicationComponent::class.java, UserInteractorModule()) {
    init {
        set { component -> appFromInstrumentation.component = component }
    }
}
